src/main/java/org/fogbowcloud/app/jes/arrebol/ArrebolJobSynchronizer.java

```diff
@@ -24,70 +23,73 @@ public ArrebolJobSynchronizer(Properties properties) {
 	@Override
 	public JDFJob synchronizeJob(JDFJob job) {
 		try {
-			String arrebolJobJson = this.requestsHelper.getJobJSON(job.getJobIdArrebol());
-			LOGGER.debug("JSON Response [" + arrebolJobJson + "]");
-			Set<ArrebolTaskState> taskStates = this.getJobTaskStates(arrebolJobJson);
-			JDFJobState jobState = getJobState(taskStates);
-			LOGGER.debug("Tasks states set [" + taskStates.toString() + "] resuming to State [" + jobState.value() + "]");
-			job.setState(jobState);
+			String arrebolJobId = job.getJobIdArrebol();
+			if(arrebolJobId != null){
+				String arrebolJobJson = this.requestsHelper.getJobJSON(arrebolJobId);
+				LOGGER.debug("JSON Response [" + arrebolJobJson + "]");
+
+				Gson gson = new Gson();
+				ArrebolJob arrebolJob = gson.fromJson(arrebolJobJson, ArrebolJob.class);
+				this.updateJob(job, arrebolJob);
+			} else {
+				LOGGER.error("ArrebolJobId from Job [" + job.getId() + "] is null.");
+			}
 		} catch (GetJobException e) {
 			LOGGER.error(e.getMessage());
 		}
 		return job;
 	}
 
-	public Set<ArrebolTaskState> getJobTaskStates(String arrebolJson) {
-		JSONObject jsonObject = new JSONObject(arrebolJson);
-		jsonObject = jsonObject.getJSONObject("tasks");
-		@SuppressWarnings("unchecked")
-		Iterator<String> keys = jsonObject.keys();
-		Set<ArrebolTaskState> taskStateList = new HashSet<ArrebolTaskState>();
-		while (keys.hasNext()) {
-			String key = keys.next();
-			JSONObject taskObject = jsonObject.getJSONObject(key);
-			String taskState = taskObject.getString("state");
-			LOGGER.debug("State [" + taskState + "] of [" + key + "] Task");
-			taskStateList.add(ArrebolTaskState.getTaskStateFromDesc(taskState));
-		}
-		return taskStateList;
+	private void updateJob(JDFJob job, ArrebolJob arrebolJob){
+		updateTasksState(job, arrebolJob);
+		updateJobState(job, arrebolJob.getJobState());
 	}
-	
-	public JDFJobState getJobState(Set<ArrebolTaskState> taskStates) {
-		if (taskStates.contains(ArrebolTaskState.FAILED)) {
-			return JDFJobState.FAILED;
-		} else if (taskStates.contains(ArrebolTaskState.RUNNING)) {
-			return JDFJobState.SUBMITTED;
-		} else if (taskStates.contains(ArrebolTaskState.FINISHED)) {
-			if(taskStates.contains(ArrebolTaskState.PENDING)) {
-				return JDFJobState.SUBMITTED;
-			} else {
-				return JDFJobState.FINISHED;
-			}
-		} else {
-			return JDFJobState.CREATED;
+
+	private void updateTasksState(JDFJob job, ArrebolJob arrebolJob){
+		Map<String, ArrebolTask> arrebolTasks = arrebolJob.getTasks();
+		LOGGER.info("Updating tasks state from job [" + job.getId() + "].");
+		for(Task task : job.getTaskList().values()){
+			ArrebolTask arrebolTask = arrebolTasks.get(task.getId());
+			ArrebolTaskState arrebolTaskState = arrebolTask.getState();
```

Comment:
I do not know if the Arrebol is returning the real Job State, @thiagomanel we still need to check the Task States to derive the Job State in Iguassu?

Reply:
keep it. As soon as I verify this feature on arrebol we decide the best way to go